sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -817,6 +818,18 @@ case class Sample(
     child: LogicalPlan)(
     val isTableSample: java.lang.Boolean = false) extends UnaryNode {
 
+  val eps = RandomSampler.roundingEpsilon
```

Comment:
I think the bounds can be generated programmatically in users' code. Any number within [0-eps, 1+eps] can be handled by RandomSampler. Also, AstBuilder.scala use eps for fraction validation as well.